script/releases/v0.5.1-rewardsv2/2-multisig.s.sol

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.12;
+
+import {MultisigCall, MultisigCallUtils, MultisigBuilder} from "zeus-templates/templates/MultisigBuilder.sol";
+import "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+import {RewardsCoordinator} from "src/contracts/core/RewardsCoordinator.sol";
+import {EigenLabsUpgrade} from "../EigenLabsUpgrade.s.sol";
+import {IPauserRegistry} from "src/contracts/interfaces/IPauserRegistry.sol";
+import {ITimelock} from "zeus-templates/interfaces/ITimelock.sol";
+import {console} from "forge-std/console.sol";
+import {EncGnosisSafe} from "zeus-templates/utils/EncGnosisSafe.sol";
+import {MultisigCallUtils, MultisigCall} from "zeus-templates/utils/MultisigCallUtils.sol";
+import {IMultiSend} from "zeus-templates/interfaces/IMultiSend.sol";
+import {TimelockController} from "@openzeppelin/contracts/governance/TimelockController.sol";
+
+/**
+ * Purpose: enqueue a multisig transaction which tells the ProxyAdmin to upgrade RewardsCoordinator. 
+ */
+contract Queue is MultisigBuilder {
+    using MultisigCallUtils for MultisigCall[];
+    using EigenLabsUpgrade for *;
+    using EncGnosisSafe for *;
+    using MultisigCallUtils for *;
+
+    MultisigCall[] private _executorCalls;
+    MultisigCall[] private _opsCalls;
+
+    function _getMultisigTransactionCalldata() internal view returns (bytes memory) {
+        ProxyAdmin pa = ProxyAdmin(this._proxyAdmin());
+
+        // ProxyAdmin::upgradeAndCall
+        bytes memory proxyAdminCalldata = abi.encodeCall(
```

Comment:
This should be `upgradeAndCall` right and not just `upgrade`

https://github.com/Layr-Labs/eigenlayer-contracts/blob/dev/script/deploy/local/Deploy_From_Scratch.s.sol#L299-L311

The proxy isn't being initialized in this case after deploy.

Reply:
Major security bug here if we don't initialize the proxy. Anyone else can initialize with whatever params they want.

Reply:
Resolved after talking offline. We just need to upgrade here since initialized during first deployment.